src/util/input-types.ts

```diff
@@ -0,0 +1,10 @@
+import {Observable, ObservableInput} from '../Observable';
+
+export type _mapProject<T, R> = (value: T, index: number) => R;
```

Comment:
I think these input types could be a separate PR as it's not a dependency for creating the *Signature interfaces.